charts/t8s-cluster/templates/management-cluster/clusterClass/clusterClass.yaml

```diff
@@ -178,7 +178,7 @@ spec:
             ref:
               apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
               kind: KubeadmConfigTemplate
-              name: {{ printf "%s-worker" $.Release.Name }}
+              name: {{ printf "%s-%s-worker" $.Release.Name ($machineDeploymentClass.flavor | contains "gpu" | ternary "gpu" "standard") }}
```

Comment:
I wouldn't say it's hard to work with, especially because we as humans don't need to interact with them.

And in debugging cases I'd prefer the more verbose name, especially as it doesn't have technical implications.